src/Microsoft.Health.Api/Registration/ApplicationBuilderExtensions.cs

```diff
@@ -37,13 +38,14 @@ public static void UseHealthChecksExtension(this IApplicationBuilder app, string
         app.UseHealthChecks(new PathString(healthCheckPathString), new HealthCheckOptions
         {
             Predicate = predicate,
-            ResponseWriter = async (httpContext, healthReport) =>
+            ResponseWriter = async (httpContext, _) =>
             {
+                HealthReport healthReport = HealthCheckPublisher.Latest;
                 var response = JsonConvert.SerializeObject(
                     new
                     {
-                        overallStatus = healthReport.Status.ToString(),
-                        details = healthReport.Entries.Select(entry => new
+                        overallStatus = healthReport == null ? HealthStatus.Unhealthy.ToString() : healthReport?.Status.ToString(),
```

Comment:
This means /health/check will return Unhealthy until `HealthCheckPublisher.Latest` is populated. Currently the delay for `HealthCheckPublisher` to be run is 20 seconds to give time for the CMK health check to return a status (~15 seconds). So /health/check will return Unhealthy for the first 20-25 seconds.

An alternative is running all the health checks when we see `HealthCheckPublisher.Latest` is null. The only weird thing here would be when CMK is down - we would consistently report status as Degraded, but then when the service/conatiner restarts, the health check would momentarily report Healthy again before the CMK background service completes its first test.

Reply:
Talked with Will about this offline; decided to instead make the calls to the cached health report async, so /health/check will wait if it doesn't have all of the data. This means the initial call will take 10-15 seconds, but subsequent calls will be very quick